cassandra/src/main/java/com/yahoo/ycsb/db/CassandraCQLClient.java

```diff
@@ -425,30 +526,34 @@ public Status insert(String table, String key,
       HashMap<String, ByteIterator> values) {
 
     try {
-      Insert insertStmt = QueryBuilder.insertInto(table);
+      final int numValues = values.size();
+      int i = 0;
+      Object[] vals = new Object[numValues + 1];
 
       // Add key
-      insertStmt.value(YCSB_KEY, key);
+      vals[i++] = key;
 
-      // Add fields
-      for (Map.Entry<String, ByteIterator> entry : values.entrySet()) {
-        Object value;
-        ByteIterator byteIterator = entry.getValue();
-        value = byteIterator.toString();
-
-        insertStmt.value(entry.getKey(), value);
+      // Add fieldList
+      for (; i < fieldList.size(); i++) {
+        String field = fieldList.get(i);
+        if (values.containsKey(field)) {
+          vals[i] = values.get(field).toString();
+        }
       }
 
-      insertStmt.setConsistencyLevel(writeConsistencyLevel);
+      // If there is one value, this is actually an update
+      BoundStatement bs = (numValues == 1 ?
+               updateStatements.get(values.keySet().iterator().next()) :
```

Comment:
Sounds good :)....

By the way, the line `is.value(YCSB_KEY, QueryBuilder.bindMarker());` seems need to  be out of the for-loop.

```java
updateStatements = new ConcurrentHashMap<String,PreparedStatement>(fieldCount);
        for (int i = 0; i < fieldCount; i++)
        {
            is = QueryBuilder.insertInto(table);
            is.value(YCSB_KEY, QueryBuilder.bindMarker());
            is.value(fieldPrefix + i, QueryBuilder.bindMarker());

            PreparedStatement ps = session.prepare(is);
            ps.setConsistencyLevel(writeConsistencyLevel);
            updateStatements.put(fieldPrefix + i, ps);
        }
```